src/main/java/uk/ac/ebi/ena/webin/cli/service/SampleService.java

```diff
@@ -47,32 +60,85 @@
         }
     }
 
-    private static class SampleResponse {
-        public int taxId;
-        public String id;
-        public String organism;
-        public String bioSampleId;
-        public boolean canBeReferenced;
+    /**
+     * Depending on what the given sample ID looks like, sample information will be retrieved from either ENA or Biosamples.
+     * Also, if the sample is not retrieved from Biosamples then an attempt will be made to retrieve it from ENA.
+     */
+    public Sample getSample(String sampleId) {
+        Sample sample = null;
+
+        if (sampleId.toUpperCase().startsWith(BIOSAMPLES_ID_PREFIX)) {
+            sample = getBiosamplesSample(sampleId);
+        }
+
+        // If, either, sample is not found on Biosamples or has incomplete metadata (e.g. Tax ID) then load the sample

```

Comment:
// If sample was not found in BioSamples or if the taxId was missing then retrieve the sample from ENA.